### DISCUSSION TITLE
form + array form + dialog

### DISCUSSION THREAD

**Discussion by Roz1c** [react-hook-form/react-hook-form#9607] [Q&A]
Hey, I'm sure this has been asked before, but...
I have the following form structure for orders:
```
{
  date: "x",
  name: "y",
  customer: "z",
  ...
  items: [
    { id: 0, name: "x", quantity: 1, ...},
  ]
}
```
If I would have editable table I would just use useForm and connect items with useFieldArray.
The problem is that items have become to complex so I have to use dialog to add/edit them.
How would I handle this with react-hook-form?

One solution would sure be to just separate the forms and submit the item in dialog, but I don't really want to do that.

**Comment by bluebill1049** [comment]
you maybe able to leverage this: https://react-hook-form.com/api/usefieldarray
